library/src-bootstrapped/scala/Tuple.scala

```diff
@@ -185,8 +194,11 @@ object Tuple {
     Tuple.fromArray(p.productIterator.toArray).asInstanceOf[m.MirroredElemTypes] // TODO use toIArray of Object to avoid double/triple array copy
 }
 
+/** Tuple of arity zero */
+sealed trait EmptyTuple extends Tuple
```

Comment:
I think this should be:
```suggestion
object EmptyTuple extends Tuple
```
And we should use EmptyTuple instead of Tuple0, unless there's some reason we need the runtime representation to differ from the compile-time one here?

Reply:
I guess one good reason is that it requires writing `EmptyTuple.type` instead of `EmptyTuple`. But then we can have both:
```suggestion
sealed trait EmptyTuple extends Tuple
object EmptyTuple extends EmptyTuple
```

Reply:
I tried that. One issue is that this gave access to the `EmptyTuple.type` which made some implicits and type matches misbehave.

Reply:
The issue is that we end up with two types for the empty tuple `EmptyTuple` and `EmptyTuple.this`.

Instead, we should simply have 
```scala
object EmptyTuple extends Tuple
```

Reply:
What about 
```
object EmptyTuple extends Tuple
type EmptyTuple = EmptyTuple.type
```